Sources/SwiftKafka/KafkaBackPressurePollingSystem.swift

```diff
@@ -0,0 +1,274 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import Logging
+import NIOConcurrencyHelpers
+import NIOCore
+
+// TODO: write test for evaluating right state transitions
```

Comment:
To make this testable instead of passing in a `KafkaClient` you can pass in a closure that polls. In real code you can pass in a closure that does the actual polling and in the test you can pass something in which you can assert on